pkg/command/root.go

```diff
@@ -307,6 +315,25 @@ func setLoggerVerbosity() {
 	}
 }
 
+func checkGlobalInit(cmd *cobra.Command) {
+	if globalinit.InitializationRequired() {
+		outStream := cmd.OutOrStderr()
+
+		fmt.Fprintf(outStream, "Some initialization of the CLI is required.\n")
+		fmt.Fprintf(outStream, "Let's set things up for you.  This will just take a few seconds.\n\n")
+
+		err := globalinit.PerformInitializations(outStream)
+		if err != nil {
+			fmt.Fprintln(outStream, "The initialization encountered an error but the CLI remains functional.")
```

Comment:
I feel a little torn about this message. 
If we can be sure that is the case, it is definitely a very useful message to show to reassure the user, given the additional error messages that are also shown. 

It is just that I am not sure with each new initializer added that we wont include some initializations that could lead to rather bad consequences if they do not run successfully.

suggestion for the future:
If we want to do this right, we might need some framework during registration, or establish some convention in the returned initializer error message to help hint at the severity of a failure and tune our message accordingly.

If we elect to pass on above for now,  another option is to be a little more conservative in the tone, or do nothing.

Reply:
I see your concern.
I did this because I was concerned about the different approaches possible:
1. silently ignore the error (with a log that only prints with increased logging) -> as you point this could leave the CLI in a bad state with no hint of why
2. completely fail execution -> this could render the CLI unusable because of a problem with an initializer
3. print the error but continue -> this may worry the user enough that they won't feel confident continuing to use the CLI

I felt 3 was the least problematic of the three choices, and to avoid the user feeling like the CLI could no longer be used, I printed "the CLI remains functional" even though that may not be the case.  I figure it is better to have the user continue to use a broken CLI and see other issues confirming the problem, than to stop using a properly functioning CLI.

> If we want to do this right, we might need some framework during registration, or establish some convention in the returned initializer error message to help hint at the severity of a failure and tune our message accordingly.

Maybe the initializer should simply not return an error if the error can be safely ignored?  Instead it could print a log itself.  Then, in `root.go` we would treat errors more seriously; still with just a printout though, not to debilitate the CLI.

Reply:
> Maybe the initializer should simply not return an error if the error can be safely ignored? Instead it could print a log itself. Then, in root.go we would treat errors more seriously; still with just a printout though, not to debilitate the CLI.

I will use this approach and print a warning if an error is returned.  For example:\
```
Some initialization of the CLI is required.
Let's set things up for you.  This will just take a few seconds.

[i] Refreshing plugin inventory cache for "projects.registry.vmware.com/tanzu_cli/plugins/plugin-inventory:latest", this will take a few seconds.
[i] Reading plugin inventory for "projects.registry.vmware.com/tanzu_cli/plugins/plugin-inventory:latest", this will take a few seconds.
[!] The initialization encountered the following error: failed to invalidate the plugin inventory cache for source

Initialization done!
==
```

Reply:
The updated approach looks reasonable as well.